rollup.config.js

```diff
@@ -633,61 +540,271 @@ function getAdapterConfig(adapterName) {
         nodeResolve({ extensions: [".ts", ".tsx"] }),
         copy({
           targets: [
-            { src: `LICENSE.md`, dest: outputDir },
-            { src: `${sourceDir}/package.json`, dest: outputDir },
-            { src: `${sourceDir}/README.md`, dest: outputDir },
+            { src: path.join(sourceDir, "LICENSE.md"), dest: outputDir },
+            { src: path.join(sourceDir, "package.json"), dest: outputDir },
+            { src: path.join(sourceDir, "README.md"), dest: outputDir },
           ],
         }),
+        magicExportsPlugin(magicExports, {
+          packageName,
+          version,
+        }),
         copyToPlaygrounds(),
       ],
     },
-    ...(hasMagicExports
-      ? [
-          {
-            external() {
-              return true;
-            },
-            input: `${sourceDir}/magicExports/remix.ts`,
-            output: {
-              banner: createBanner(packageName, version),
-              dir: `${outputDir}/magicExports`,
-              format: "cjs",
-            },
-            plugins: [
-              babel({
-                babelHelpers: "bundled",
-                exclude: /node_modules/,
-                extensions: [".ts", ".tsx"],
-              }),
-              copyToPlaygrounds(),
-            ],
-          },
-          {
-            external() {
-              return true;
-            },
-            input: `${sourceDir}/magicExports/remix.ts`,
-            output: {
-              banner: createBanner(packageName, version),
-              dir: `${outputDir}/magicExports/esm`,
-              format: "esm",
-            },
-            plugins: [
-              babel({
-                babelHelpers: "bundled",
-                exclude: /node_modules/,
-                extensions: [".ts", ".tsx"],
-              }),
-              copyToPlaygrounds(),
-            ],
-          },
-        ]
-      : []),
   ];
 }
 
+/**
+ * TODO: Remove in v2
+ * @param {RemixPackage} packageName
+ * @returns {MagicExports | null}
+ */
+function getMagicExports(packageName) {
+  // Re-export everything from packages that is available in `remix`
+  /** @type {MagicExports | null} */
+  let magicExports = null;
+  switch (packageName) {
+    case "@remix-run/architect":
+      magicExports = {
+        values: {
+          "@remix-run/architect": ["createArcTableSessionStorage"],
+        },
+        types: {
+          "@remix-run/node": ["UploadHandler", "UploadHandlerPart"],
+        },
+      };
+      break;
+    case "@remix-run/cloudflare-pages":
+      magicExports = {
+        values: {
+          "@remix-run/cloudflare": ["createCloudflareKVSessionStorage"],
+        },
+      };
+      break;
```

Comment:
I don't think this one is necessary, because we already have this exported by `@remix-run/cloudflare`

```suggestion
```